### PR TITLE
Use bintray to host Appium zip builds

### PR DESCRIPTION
**Pull request by dpgraham**
* Upload a zip every commit that can be used for doing staging tests on Sauce Labs
* Added a script that zips up the Appium folder
* Added a script that uploads the zip to bintray
* Made a Travis build that does a production build, transpiles source code, prunes the packages and then zips and uploads it

# Other Changes
* Upgraded node versions to 7 and 8 from 6 and 7
* Use the new Travis CI 'jobs'

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,5 +8,35 @@ addons:
...
     packages:
       - g++-4.8
-script:
-  - npm run generate-docs && npm run test && npm run e2e-test
+jobs:
+  include:
+    - name: test
+      node_js:
+        - "8"
+        - "7"
```
Comment (imurchie) [inline .travis.yml]: Node 7 is EOLed. We should be testing on 6 and 8 at the moment.

File: .travis.yml [hunk 2]
```diff
@@ -11,5 +8,35 @@ addons:
...
+      script:
+        # Install prod dependencies
+        - npm install
+
+        # Install only the dev dependencies that we need
+        - npm install --only=dev gulp
+        - npm install --only=dev appium-gulp-plugins
+
+        # Build the assets
+        - $(npm bin)/gulp transpile
```
Comment (imurchie) [inline .travis.yml]: Does `npm run build` not work?

**Comment by imurchie** [comment]
Weird. That last push did not trigger a build

File: test/scripts/bintray-upload.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+const { logger } = require('appium-support');
+const path = require('path');
+const fs = require('fs');
+
+let log = logger.getLogger('Bintray');
+
+(async function () {
+
+  // Bintray info
+  const BINTRAY_USERNAME = process.env.BINTRAY_USERNAME;
```
Comment (jlipps) [inline test/scripts/bintray-upload.js]: why not put these constants at the top of the file outside this function?

**Review by jlipps — APPROVED** [review]

**Pull request merged**